.devcontainer/Dockerfile.llvm-project

```diff
@@ -0,0 +1,131 @@
+# By default, use ubuntu:focal
+ARG BASE_IMAGE="ubuntu:focal"
+FROM ${BASE_IMAGE}
+
+# Label the image for various checking and cleanup
+ARG LLVM_PROJECT_SHA1="ba8b8a73fcb6b830e63cd8e20c6e13b2a14d69bf"
+LABEL llvm_project_sha1=${LLVM_PROJECT_SHA1}
+
+
+ARG NPROC=16
+ARG WORK_DIR=/workdir
+WORKDIR ${WORK_DIR}
+
+# Install tools needed
+RUN distro=$(cat /etc/os-release|grep -Po '(?<=^ID=").*(?=")|(?<=^ID=)[^"].*[^"]') \
+    && TZ="Europe/Berlin" \
+    && if [ "${distro}" = "debian" ] || [ "${distro}" = "ubuntu" ]; then \
+          DEBIAN_FRONTEND=noninteractive && \
+          apt-get update -qq && \
+          apt-get install -qq -y --no-install-recommends tzdata && \
+          ln -sf /usr/share/zoneinfo/${TZ} /etc/localtime && \
+          dpkg-reconfigure -f noninteractive tzdata && \
+          apt-get install -qq -y --no-install-recommends \
+                  autoconf automake ca-certificates cmake curl \
+                  default-jdk-headless gcc g++ git libncurses-dev \
+                  libtool make maven ninja-build openjdk-11-jdk-headless \
+                  python3 python3-dev python3-distutils python3-numpy \
+                  python3-pip python3-pytest-xdist python3-setuptools \
+                  python3-typing-extensions zlib1g-dev && \
+          rm -rf /var/lib/apt/lists/* && \
+          ln -sf /usr/bin/pytest-3 /usr/bin/pytest; \
+       elif [ "${distro}" = "rhel" ] || [ "${distro}" = "fedora" ]; then \
+          ln -sf /usr/share/zoneinfo/${TZ} /etc/localtime && \
+	  ([ -x /usr/bin/microdnf ] && microdnf install -y yum) && \
+          yum install -q -y \
+              https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm && \
+          yum update -q -y && \
+          yum install -q -y \
+              autoconf automake ca-certificates cmake diffutils \
+              file java-11-openjdk-devel java-11-openjdk-headless \
+              gcc gcc-c++ git libtool make ncurses-devel \
+              python39 python39-devel python39-numpy python39-pip \
+              python39-setuptools python39-wheel zlib-devel && \
+          # Use same versions as those in ubuntu:focal
+          pip3 install -q \
+               Cython pytest==4.6.9 pytest-forked==1.1.3 \
+               pytest-xdist==1.31.0 typing-extensions==3.7.4.1 && \
+          rm -rf /var/cache/dnf/* && \
+          # Install ninja
+          git clone -b v1.10.2 https://github.com/ninja-build/ninja.git && \
+          cd ninja && mkdir -p build && cd build && \
+          cmake .. && \
+          make -j${NPROC} install && \
+          cd ${WORK_DIR} && rm -rf ninja; \
+       fi \
+    && ln -sf /usr/bin/python3 /usr/bin/python
+
+
+RUN apt-get update
```

Comment:
It would be great if you can merge most of the apt-get install commands. Each RUN, ADD, COPY command in a Docker file is a "layer" and each layer is cached. 

That means if one of the early lines changes, then all subsequent layers are rebuilt.